docs/src/index.md

```diff
@@ -1,8 +1,131 @@
+and untar it somewhere.
+
+```bash
+cd ~
+wget https://github.com/microsoft/LightGBM/archive/v2.3.1.tar.gz
+tar -xf v2.3.1.tar.gz
+```
+
+```julia
+using LightGBM
+using DelimitedFiles
+
+LIGHTGBM_SOURCE = abspath("~/LightGBM-2.3.1")
+
+# Load LightGBM's binary classification example.
+binary_test = readdlm(joinpath(LIGHTGBM_SOURCE, "examples", "binary_classification", "binary.test"), '\t')
+binary_train = readdlm(joinpath(LIGHTGBM_SOURCE, "examples", "binary_classification", "binary.train"), '\t')
+X_train = binary_train[:, 2:end]
+y_train = binary_train[:, 1]
+X_test = binary_test[:, 2:end]
+y_test = binary_test[:, 1]
+
+# Create an estimator with the desired parameters—leave other parameters at the default values.
+estimator = LGBMClassification(
+    objective = "binary",
+    num_iterations = 100,
+    learning_rate = .1,
+    early_stopping_round = 5,
+    feature_fraction = .8,
+    bagging_fraction = .9,
+    bagging_freq = 1,
+    num_leaves = 1000,
+    num_class = 1,
+    metric = ["auc", "binary_logloss"]
+)
+
+# Fit the estimator on the training data and return its scores for the test data.
+fit!(estimator, X_train, y_train, (X_test, y_test))
+
+# Predict arbitrary data with the estimator.
+predict(estimator, X_train)
+
+# Cross-validate using a two-fold cross-validation iterable providing training indices.
+splits = (collect(1:3500), collect(3501:7000))
+cv(estimator, X_train, y_train, splits)
+
+# Exhaustive search on an iterable containing all combinations of learning_rate ∈ {.1, .2} and
+# bagging_fraction ∈ {.8, .9}
+params = [Dict(:learning_rate => learning_rate,
+               :bagging_fraction => bagging_fraction) for
+          learning_rate in (.1, .2),
+          bagging_fraction in (.8, .9)]
+search_cv(estimator, X_train, y_train, splits, params)
+
+# Save and load the fitted model.
+filename = pwd() * "/finished.model"
+savemodel(estimator, filename)
+loadmodel(estimator, filename)
 ```
 
-```@autodocs
-Modules = [LightGBM]
+# Exports
+
+Note that a lot of parameters used within this module and in the code and examples are
+exact matches with those from [LightGBM.](https://lightgbm.readthedocs.io/en/latest/Parameters.html)
+Not all of these are necessarily supported but see the guide for detailed explanations of what these
+parameters do and their valid values.
+
+# MLJ Support
+
+This package has an interface to [MLJ](https://github.com/alan-turing-institute/MLJ.jl).
+Exhaustive MLJ documentation is out of scope for here, however the main things are:
+
+The MLJ interface models are
+```julia
+LightGBM.MLJInterface.LGBMClassifier
+LightGBM.MLJInterface.LGBMRegressor
 ```
+
+And these have the same interface parameters as the [estimators](#estimators)
+
+The interface models are generally passed to `MLJBase.fit` or `MLJBase.machine`
+and integrated as part of a larger MLJ pipeline. [An example is provided](https://alan-turing-institute.github.io/MLJTutorials/end-to-end/boston-lgbm/)
```

Comment:
The link is broken it needs updating to this: (that's not your fault)

https://alan-turing-institute.github.io/DataScienceTutorials.jl/end-to-end/boston-lgbm/

Reply:
The link was broken and was updated as follows (https://alan-turing-institute.github.io/DataScienceTutorials.jl/end-to-end/boston-lgbm/)